inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -435,3 +464,40 @@ def has_more_than_one_exact_match(obj, eng):
     """Does the record have more than one exact match."""
     exact_matches = obj.extra_data['matches']['exact']
     return len(set(exact_matches)) > 1
+
+
+@with_debug_logging
+def run_next_if_necessary(obj, eng):
+    """Remove current wf id from matched workflows and run next one"""
+    blocked_wfs = set_wf_not_completed_ids_to_wf(obj, False)
```

Comment:
I would use kwargs, it's not very intuitive which flag you are setting to False from this call:
`set_wf_not_completed_ids_to_wf(obj, skip_blocked=False, skip_halted=False)`